templates/nft-minting-dapp-template/frontend/hooks/useGetCollectionData.ts

```diff
@@ -79,15 +75,45 @@ async function getStartAndEndTime(collection_id: string): Promise<[start: Date,
   ];
 }
 
+async function getUserMintBalance(
+  user_address: string,
+  collection_address: string,
+  stage_name: string,
+): Promise<number> {
+  const userMintedAmount = await aptosClient().view<[string]>({
+    payload: {
+      function: `${AccountAddress.from(MODULE_ADDRESS)}::launchpad::get_mint_balance`,
+      functionArguments: [collection_address, stage_name, user_address],
+    },
+  });
+
+  return Number(userMintedAmount[0]);
+}
+
 export function useGetCollectionData(collection_id: string = config.collection_id) {
+  const { account } = useWallet();
+
   return useQuery({
     queryKey: ["app-state", collection_id],
     refetchInterval: 1000 * 30,
     queryFn: async () => {
       try {
         if (!collection_id) return null;
 
-        const [startDate, endDate, isMintInfinite] = await getStartAndEndTime(collection_id);
+        const mintStageRes = await aptosClient().view<[{ vec: [string] | [] }]>({
+          payload: {
+            function: `${AccountAddress.from(MODULE_ADDRESS)}::launchpad::get_active_or_next_mint_stage`,
+            functionArguments: [collection_id],
+          },
+        });
+
+        if (mintStageRes[0].vec.length === 0) return null;
```

Comment:
a bit confused - is this a check on whether there is an active mint, i.e a collection has been made, and if there is isn't we wont do the indexer call?

Reply:
updated, previous code will throw error when mint stage not found, now ill only return collection data if mint stage not found